shell/plugins/steve/subscribe.js

```diff
@@ -491,6 +576,24 @@ const sharedActions = {
     }
   },
 
+  /**
+   * Unwatch watches that are incompatible with the new type
```

Comment:
Why wouldn't we do an early exit in watch to prevent the type from ever being watched?

Reply:
this covers the case where a user switches between resource findXs. the new watch request is the one we want to keep, and incompatible ones we want to stop
 
- loads a detail page (`find` + watch one resource) --> list page (`findPage` + watch for resource.changes) 
- loads a page that uses `findAll` (watch all resources) --> page that uses `findPage` (watch for resource.changes) --> original page `findAll` (watch all resources)